app/services/path_generator_service.rb

```diff
@@ -1,33 +1,35 @@
 # frozen_string_literal: true
 
-class PathGeneratorService
-  class ErrorGeneratingPath < RuntimeError
-  end
-
-  def initialize(max_repeated_titles = 1000)
+class PathGeneratorService < ApplicationService
+  def initialize(document, proposed_title, max_repeated_titles: 1000)
+    @document = document
+    @proposed_title = proposed_title
     @max_repeated_titles = max_repeated_titles
   end
 
-  def path(document, proposed_title)
+  def call
     prefix = document.document_type.path_prefix
     slug = proposed_title.parameterize
     base_path = create_path(prefix, slug)
-    return base_path unless document_exists_with_path?(base_path, document)
+    return base_path unless path_in_use?(base_path)
 
-    find_a_unique_path(document, prefix, slug)
+    find_a_unique_path(prefix, slug)
   end
 
 private
 
-  def find_a_unique_path(document, prefix, slug)
-    (1..@max_repeated_titles).each do |appended_count|
+  attr_reader :document, :proposed_title, :max_repeated_titles
+
+  def find_a_unique_path(prefix, slug)
+    (1..max_repeated_titles).each do |appended_count|
       base_path = create_path(prefix, slug, appended_count)
-      return base_path unless document_exists_with_path?(base_path, document)
+      return base_path unless path_in_use?(base_path)
     end
-    raise(ErrorGeneratingPath, "Already >#{@max_repeated_titles} paths with same title.")
+
+    raise "Already >#{max_repeated_titles} paths with same title."
```

Comment:
I guess the fact that this can switch to a RuntimeError means that nothing rescues this error 🙈

Reply:
Indeed. I thought this would be more consistent with how e do this elsewhere.